README.md

```diff
@@ -31,129 +34,121 @@ $ npm install react-plotly.js plotly.js
-        {
-          type: 'bar',
-          x: [1, 2, 3],
-          y: [2, 5, 3]
-        }
-      ]}
-
-      layout={{
-        width: 320,
-        height: 240,
-        title: 'A Fancy Plot'
-      }}
-    />
-  );
+class App extends React.Component {
+  render() {
+    return (
+      <Plot
+        data={[
+          {
+            x: [1, 2, 3],
+            y: [2, 6, 3],
+            type: 'scatter',
+            mode: 'lines+points',
+            marker: {color: 'red'},
+          },
+          {type: 'bar', x: [1, 2, 3], y: [2, 5, 3]},
+        ]}
+        layout={{width: 320, height: 240, title: 'A Fancy Plot'}}
+      />
+    );
+  }
 }
 ```
 
 You should see a plot like this:
 
 <p align="center">
-  <img src="example.png" alt="Example plot" width="320" height="240">
+ <img src="example.png" alt="Example plot" width="320" height="240">
 </p>
 
 For a full description of Plotly chart types and attributes see the following resources:
 
 * [Plotly JavaScript API documentation](https://plot.ly/javascript/)
 * [Full plotly.js attribute listing](https://plot.ly/javascript/reference/)
 
-## Customizing the `plotly.js` bundle
+## State management
 
-By default, the `Plot` component exported by this library loads a precompiled version of all of `plotly.js`, so `plotly.js` must be installed as a peer dependency. This bundle is around 6Mb unminified, and minifies to just over 2Mb.
+This is a "dumb" component with no internal state. This means that if a user interacts with the plot, by zooming or panning for example, any subsequent re-renders will lose this information unless it is captured and upstreamed via the `onUpdate` callback prop.
 
-If you do not wish to use this version of `plotly.js`, e.g. if you want to use a [different precompiled bundle](https://github.com/plotly/plotly.js/blob/master/dist/README.md#partial-bundles) or if your wish to [assemble you own customized bundle](https://github.com/plotly/plotly.js#modules), or if you wish to load `plotly.js` [from a CDN](https://github.com/plotly/plotly.js#use-the-plotlyjs-cdn-hosted-by-fastly), you can skip the installation of as a peer dependency (and ignore the resulting warning) and use the `createPlotComponent` method to get a `Plot` component, instead of importing it:
+Here is a simple example of how to capture and store state in a parent object:
 
 ```javascript
-// simplest method: uses precompiled complete bundle from `plotly.js`
-import Plot from 'react-plotly.js';
-
-// customizable method: use your own `Plotly` object
-import createPlotlyComponent from 'react-plotly.js/factory';
-const Plot = createPlotlyComponent(Plotly);
+class App extends React.Component {
+    Plot(props) {
+        super(props);
+        this.state = {
+          /* data, layout, config, frames */
+        };
+    }
+
+    render() {
+        return (
+            <Plot
+                data={this.state.data}
+                layout={this.state.layout}
+                frames={this.state.frames}
```

Comment:
so here it'd be this.state.figure.frames, this.state.figure.layout, this.state.figure.layout..right

Reply:
nope, `this.setState(figure)` means that the state *is* the figure. You would be right if I'd done `this.setState({figure})` (note the `{ }`).